tensorflow-framework/src/main/java/org/tensorflow/framework/op/FrameworkOps.java

```diff
@@ -0,0 +1,165 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.op;
+
+import org.tensorflow.DeviceSpec;
+import org.tensorflow.EagerSession;
+import org.tensorflow.ExecutionEnvironment;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.Scope;
+
+/**
+ * An API for building framework operations as {@link Op Op}s
+ *
+ * <p>These are higher level ops that may invoke core ops. Higher level Ops may perform the
+ * operation solely in the TensorFlow framework or do preprocessing of the Operands before invoking
+ * a core level Op.
+ */
+public class FrameworkOps {
+  public final Ops coreOps;
+  public final NnOps nn;
+  public final SetOps sets;
+  public final MathOps math;
+  public final LinalgOps linalg;
+  private final Scope scope;
+
+  /**
+   * Creates a FrameworkOps instance with the provided scope
+   *
+   * @param scope the scope
+   */
+  private FrameworkOps(Scope scope) {
+    this.coreOps = Ops.create(scope.env());
+    this.scope = scope;
+    nn = new NnOps(this);
+    sets = new SetOps(this);
+    math = new MathOps(this);
+    linalg = new LinalgOps(this);
+  }
+
+  /**
+   * Creates a FrameworkOps instance based on the provided Core Ops
+   *
+   * @param coreOps The TensorFlow Core Ops
+   */
+  private FrameworkOps(Ops coreOps) {
+    this.coreOps = coreOps;
+    this.scope = coreOps.scope();
+    nn = new NnOps(this);
+    sets = new SetOps(this);
+    math = new MathOps(this);
+    linalg = new LinalgOps(this);
+
+  }
+
+  /**
+   * Creates an API for building operations in the provided execution environment
+   *
+   * @param env the exection environment
+   * @return the FrameworkOps
+   */
+  public static FrameworkOps create(ExecutionEnvironment env) {
+    return new FrameworkOps(new Scope(env));
+  }
+
+  /**
+   * Creates an API for building operations in the default eager execution environment
+   *
+   * <p>Invoking this method is equivalent to {@code
+   * FrameworkOps.create(EagerSession.getDefault())}.
+   *
+   * @return the FrameworkOps
+   */
+  public static FrameworkOps create() {
+    return new FrameworkOps(new Scope(EagerSession.getDefault()));
+  }
+
+  /**
+   * Creates an API for building operations in the default eager execution environment
+   *
+   * @param coreOps the TensorFlow core Ops
+   * @return the FrameworkOps
+   */
+  public static FrameworkOps create(Ops coreOps) {
+    return new FrameworkOps(coreOps);
+  }
+
+  /**
+   * Returns the current {@link Scope scope} of this API
+   *
+   * @return the current {@link Scope scope} of this API
+   */
+  public final Scope scope() {
+    return scope;
+  }
+
+  /**
+   * Gets the core Ops
+   *
+   * @return coreOps
+   */
+  public final Ops coreOps() {
+    return coreOps;
```

Comment:
do you really need a getter since `coreOps` is public? Also, can we rename to `raw` maybe? so you can do:
```java
FrameworkOps ftf = FrameworkOps.create(tf);
ftf.nn.sparseSoftmaxCrossEntropyWithLogits(...)
ftf.raw.nn.sparseSoftmaxCrossEntropyWithLogits(...)
```

Reply:
@karllessard I don't like `raw`, how about `core`? ftf.core.nn.sparseSoftmaxCrossEntropyWithLogits(...)